src/js/angular/aclmanagement/controllers.js

```diff
@@ -328,14 +329,11 @@ function AclManagementCtrl($scope, $location, toastr, AclManagementRestService,
      * @param {FormController} form - The form object.
      */
     $scope.performSearchActionOnEnter = function (event, scope, form) {
-        if (!RoleNamePrefixUtils.isCustomPrefixUsed(event.target.value)) {
-            $scope.showPrefixWarningIcon = false;
-        }
         if (event.keyCode === 13) {
             event.stopPropagation();
             event.preventDefault();
             $scope.triggerValidation(form);
-            if (form.$valid) {
+            if (!form.$error.required || !form.$error.minLength) {
```

Comment:
On a second thought, the idea to have the prefix validation logic in the directive doesn't look like a good idea. The problem now is that because of the new validation rule you make the model dirty and invalid, which enforces you to do the stuff like the one above. And I  don't like it. I someone adds a new validator someday, and that validator needs to make the model invalid in certain cases, then he/she must know that the code should also be changed somewhere else in order for the feature to work as expected. In other words, the specific new validator needs to be added here in this statement. This is bad because it couples the directive with the controller.

Reply:
In my oppinion, if this new validation logic related with the custom_ prefix stays in the directive, then it should not change the model validity, but rather put some additional property like $warning instead and the UI logic (in the template) to use it instead. Right now, the field is highlighted as invalid, although according to the requirement, the presence of the custom_ prefix should trigger just a warning to appear.